src/aspire/utils/preprocess.py

```diff
@@ -302,3 +303,71 @@ def vol2img(volume, rots, L=None, dtype=None):
         im = im * m_reshape(np.exp(2*np.pi*1j*(grid2d['x'] +grid2d['y'])/(2*lv)), (lv, lv, 1))
 
     return np.real(im)
+
+
+def fuzzy_mask(n, dims, r0, risetime, origin=None):
+    """
+    Create a centered 1 to 3d object of radius r0
+
+    Made with an error function with effective rise time.
+    :param n:
+    :param dims:
+    :param r0:
+    :param risetime:
+    :param origin:
+    :return:
+    """
+
+    if isinstance(n, int):
+        n = np.array([n])
+
+    if isinstance(r0, int):
+        r0 = np.array([r0])
+
+    center = (n + 1.0) / 2
+    k = 1.782 / risetime
+
+    if dims == 1:
```

Comment:
Splited the fuzzy_mask function into three and removed the dimensional check in the commit 3c5ec71.

Reply:
That's not having one function independent of the dimension. It's still duplicating code and moving the if statements somewhere else.

We should have a *single* function that takes a `dims` argument (or optionally, checks the length of `L` and takes this to be `dims`). It can then generate the grids vectorially, like this
```python
grids = [np.arange(-ell // 2, ell // 2) for ell in L]
XYZ = np.meshgrid(*grids, indexing='ij')
XYZ_sq = [X ** 2 for X in XYZ]
R = np.sqrt(np.add(*XYZ_sq))
# or better
R = np.hypot(*XYZ)
```
and so on. This way, we know the code is doing the same thing for all dimensions, it's easier to read, and easier to maintain.

Reply:
Rewrote in commit f1a4add. We might come back for finalizing preprocess steps.